### PR TITLE
feat: upgrade to new ipfs-block and blockservice

### PR DESCRIPTION
**Pull request by dignifiedquire**
Ref https://github.com/ipfs/js-ipfs/issues/787

### CODE REVIEW

File: test/dag-node-test.js [hunk 1]
```diff
@@ -406,51 +411,29 @@ module.exports = (repo) => {
...
           }
         ], done)
       })
     })
 
     it('deserialize go-ipfs block from ipldResolver', (done) => {
-      const bs = new BlockService(repo)
+      if (!isNode) {
+        return done()
+      }
```
Comment (diasdavid) [inline test/dag-node-test.js]: Why remove this test from the browser test suit?

File: test/browser.js [hunk 2]
```diff
@@ -1,58 +1,42 @@
...
-      )
-    }), done)
+    ], done)
   })
 
-  const repo = new IPFSRepo(path, {stores: Store})
-
   require('./dag-node-test')(repo)
   require('./dag-link-test')(repo)
 })
```
Comment (diasdavid) [inline test/browser.js]: This browser.js was also setting up the repo with some blocks for tests.

Comment (dignifiedquire) [inline test/browser.js]: Remember when I said we turn sharding off in the browser? That's the reason that setup doesn't work anymore.

File: test/dag-node-test.js (hunk 1 above)
Comment (dignifiedquire) [inline test/dag-node-test.js]: See below

File: test/browser.js (hunk 2 above)
Comment (diasdavid) [inline test/browser.js]: roger :)

**Pull request merged**